### PR TITLE
Flying text

### PR DESCRIPTION
**Pull request by Nexela**
Might be easier to look at on a commit by commit basis, The empty spaces in lines where deep!

### CODE REVIEW

File: control.lua [hunk 1]
```diff
@@ -276,19 +276,20 @@ local function find_note( ent )
...
-    end
+    local invis_note = ent.surface.find_entities_filtered{name="invis-note",area={{x-0.01,y-0.01},{x+0.01, y+0.01}}}[1]
+	game.print(serpent.line(invis_note, {comment=false}))
+    --local invis_note
+    -- for _, entity in ipairs(entities) do
+    --     if entity.position.x==x and entity.position.y==y then
+    --         invis_note = entity
+    --         break
+    --     end
+    -- end
```
Comment (NiftyManiac) [inline control.lua]: The point of the loop was to ensure we don't get an entity with a slight offset. Just in case someone was doing trickery with minor offsets.

File: control.lua [hunk 2]
```diff
@@ -276,9 +276,7 @@ local function find_note( ent )
     -- note: find_entity doesn't seem to work when the collision box is 0
     local x = ent.position.x
     local y = ent.position.y
-    local invis_note = ent.surface.find_entities_filtered{name="invis-note",area={{x-0.01,y-0.01},{x+0.01, y+0.01}}}[1]
-	--local invis_note = ent.surface.find_entities_filtered{name="invis-note", position=ent.position}[1]
-
+    local invis_note = ent.surface.find_entities_filtered{name="invis-note",area={{x-0.1,y-0.1},{x+0.1, y+0.1}}}[1]
```
Comment (NiftyManiac) [inline control.lua]: Why the change?

File: control.lua [hunk 3]
```diff
@@ -485,9 +485,12 @@ local function on_creation( event )
     local ent = event.created_entity
     debug_print( "on_creation ", ent.name )
 
+	-- There is a bug here with this logic to test it out create a blueprint and click drag it,
+	-- it will make lots of invis notes.  -- I have a few ideas on how to fix it but will need time to test
+
```
Comment (NiftyManiac) [inline control.lua]: Put in issues

File: control.lua (hunk 1 above)
Comment (NiftyManiac) [reply control.lua]: Honestly we should probably encode the target entity name in the note too, otherwise it's impossible to avoid sticking the note on trees or other entities that happen to be in the same place.

Comment (NiftyManiac) [reply control.lua]: Huh, just ran into the bug that you might have been trying to fix here, where invis-notes are placed imprecisely. Wonder where that comes from.

File: control.lua (hunk 2 above)
Comment (Nexela) [reply control.lua]: I Was playing around with some things and didn't set it back to the correct one.

File: control.lua (hunk 1 above)
Comment (Nexela) [reply control.lua]: Totally missed all of this from early and I am on the verge of passing out. 👍 

I will get another chance to work on this tomorrow night as I have some ideas to will probably fix both bugs hopefully.
First one linking by unit numbers,  For the blueprint placing, make sure the invis note is at ghost position before reviving else delete
